RELEASE.md

```diff
@@ -0,0 +1,31 @@
+# Explaination of the versionning scheme
+
+OCamlformat now an unsual versionning scheme. For every version `X.Y.Z` of ocamlformat,
+there are multiple packages :
+
+- `ocamlformat.X.Y.Z` which contains the `ocamlformat` executable
+- `ocamlformat-X_Y_Z.1.0` which contains the `ocamformat-X_Y_Z` executable
+
+You also have packages for the libs : `ocamlformat-lib.X.Y.Z` and
```

Comment:
Out of curiosity, why do you need to do this for the library? Are you expecting to co-install multiple libraries of ocamlformat-lib in the same switch? Who are the consumers of those libraries?

Reply:
The consumers of the `ocamlformat-lib-X_Y_Z` package are `ocamlformat-lib.X.Y.Z` and `ocamlformat-X_Y_Z.1.0`.

The executable needs it for obvious reasons, and then we want to package the lib for general use, as it has been packaged before, under the name `ocamlformat-lib`.
This felt like the simplest way to make this lib accessible to both packages. The versionned lib itself is not meant to be used by itself. An alternative would to have two equivalent libraries with a dune rule that copy files. Then the `ocamlformat-lib-X_Y_Z` would be private.

I do not really like copying source files around like that (especially because its not all in one folder, the vendor folder is also part of the `ocamlformat-lib-X_Y_Z package), but what we have now is not ideal either. I will think more about and I am happy to receive input on that topic.

Reply:
I am a bit confused - I though the `ocamlformat-X_Y_Z` would only contain the `ocamlfomat` binary and be self-contained (ie. without external dependencies, so you could easily co-install all of these packages in the same switch without dependency conflicts). If you do this, why would they need to depend on `ocamlformat-lib-X_Y_Z`. If nobody is using `ocamlformat-lib_X_Y_Z` I would just not build more maintenance burden by releasing them :-)

Reply:
`ocamlformat-X_Y_Z` has a real dependency on `ocamlformat-lib-X_Y_Z`, thats how it is implemented. We might do vendoring to hide it, but this will require duplicating sources (maybe this is actually fine ?)
I will try to find a satisfying solution for this, and in the meantime, I have removed the separate package (now you have `ocamlformat-X_Y_Z.ocamlformat-lib`).
As for dependencies, I think you are right, and we should vendor them. It would be very hard to remove them for all ocamlformat previous versions, but vendoring should work and make sure everything always work. It is going to be a bit more work for each version but I think its worth it.
For forward versions we will try to remove dependencies, especially on base.